legal-api/src/legal_api/services/filings/validations/change_of_directors.py

```diff
@@ -70,43 +70,40 @@ def validate_directors_addresses(cod: Dict) -> List:
 def validate_effective_date(business: Business, cod: Dict) -> List:
     """Return an error or warning message based on the effective date validation rules.
 
-    Rules:
+    Rules: (text from the BA rules document)
         - The effective date of change cannot be in the future.
-        - The effective date cannot be a date prior to their Incorporation Date
+        - The effective date cannot be a date prior to their Incorporation Date.
         - The effective date of change cannot be a date that is farther in the past
-            as a previous COD filing (Standalone or AR).
+            than a previous COD filing (standalone or AR).
         - The effective date can be the same effective date as another COD filing
-            (standalone OR AR). If this is the case:
-        - COD filing that was filed most recently as the most current director information.
+            (standalone or AR). If this is the case:
+        - COD filing that was filed most recently is the most current director information.
     """
     try:
+        # NB: COD effective date is at 0 hours UTC.
         filing_effective_date = cod['filing']['header']['effectiveDate']
     except KeyError:
-        try:
-            # we'll assume the filing is at 0 hours UTC
-            filing_effective_date = cod['filing']['header']['date'] + 'T00:00:00+00:00'
-        except KeyError:
-            return {'error': babel('No effective_date or filing date provided.')}
+        return {'error': babel('No effective_date provided.')}
 
     try:
         effective_date = datetime.fromisoformat(filing_effective_date)
     except ValueError:
-        return {'error': babel('Invalid ISO format for effective_date or filing date.')}
+        return {'error': babel('Invalid ISO format for effective_date.')}
 
     msg = []
 
-    # The effective date of change cannot be in the future
+    # The effective datetime cannot be in the future.
     if effective_date > datetime.utcnow():
         msg.append({'error': babel('Filing cannot have a future effective date.')})
 
-    # The effective date cannot be a date prior to their Incorporation Date
-    if effective_date < business.founding_date:
-        msg.append({'error': babel('Filing cannot be before a businesses founding date.')})
+    # The effective date cannot be before their Incorporation Date.
+    if effective_date.date() < business.founding_date.date():
```

Comment:
this would not allow this filing on the same date as incorporation, which it can be. It just has to be later then the date + time it was incoporated.

Reply:
This code rejects the filing if it is before the incorp date, so it allows this filing on the same day as the incorp date. Or am I confused?
Also, there is no effective time -- just a date with time = 00:00:00.000. So we can only check dates,

Reply:
founding date is a datetime, for new IAs it has the time and other filings can be on the same day, but later than incorp time.

Reply:
You're right, but:
- this is now old code
- there is no time for a COD filing -- just a date -- so we can't check whether the filing is before the incorp time; we can only check date

Reply:
the submission is a datetime

Reply:
OK, I am adding a check that the submission datetime (ie, "date" provided by the UI) is not less than the business founding date. This is thus a sanity check to validate that the UI is not providing corrupt data.

However, I cannot change the **effective datetime** of a COD -- it is 00:00 local on the date selected, which could be the same date they incorporated (but the COD **submission datetime** will be >= incorporation datetime as long as UI/BE times are close together and valid data is sent across).